lib/view_component/collection.rb

```diff
@@ -35,5 +36,18 @@ def component_options(item, iterator)
 
       @options.merge(item_options)
     end
+
+    def ensure_initializer_accepts_iterator
+      # If the component does not set a custom collection parameter,
+      # make sure the default parameter is accepted by the
+      # component initializer.
+      if !@component.with_collection_parameter_attr &&
+        !@component.instance_method(:initialize).parameters.map(&:last).include?(@component.collection_parameter_name)
+        raise ArgumentError.new(
+          "#{@component} initializer must accept " \
+          "`#{@component.collection_parameter_name}` collection parameter."
+        )
```

Comment:
Can we perhaps DRY up the implementation a little here? Can't we just make this a method on the component

```ruby
collection_param_error = @component.validate_collection_parameter!(default_only: true)
raise ArgumentError.new(collection_param_error) if collection_param_error
```

that way we can avoid the copy-past of the `instance_method(:initialize)` check and the construction of the error string